### PR TITLE
spr 1.3.4 (new formula)

### PR DESCRIPTION
**Pull request by sven-of-cord**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This is a new version of #105544, with a new release of the software and added tests in the formula. Thanks for the feedback on the previous PR, @carlocab !

### CODE REVIEW

File: Formula/spr.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
+class Spr < Formula
+  desc "Submit pull requests for individual, amendable, rebaseable commits to GitHub"
+  homepage "https://github.com/getcord/spr"
+  url "https://github.com/getcord/spr.git",
+    tag:      "v1.3.4",
+    revision: "1b05ef36f0beb0ff8bdb376bf98ce77a41e41605"
```
Comment (carlocab) [inline Formula/spr.rb:6]: Align the arguments of `url`, please, like so:
```suggestion
  url "https://github.com/getcord/spr.git",
      tag:      "v1.3.4",
      revision: "1b05ef36f0beb0ff8bdb376bf98ce77a41e41605"
```
Though, do we really need a Git clone? Can we use a tarball instead?

File: Formula/spr.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  test do
+    spr = "#{bin}/spr"
+    assert_match "spr #{version}", shell_output("#{spr} --version")
+
+    system "git", "init", "-b", "trunk", testpath/"test-repo"
+    system "git", "config", "--global", "user.email", "nobody@example.com"
+    system "git", "config", "--global", "user.name", "Nobody"
+    cd testpath/"test-repo" do
```
Comment (carlocab) [inline Formula/spr.rb:27]:
```suggestion
    cd "test-repo" do
```

File: Formula/spr.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      system "git", "config", "spr.githubMasterBranch", "trunk"
+
+      # Some bogus config
+      system "git", "config", "spr.githubRepository", "a/b"
+      system "git", "config", "spr.branchPrefix", "spr/"
+
+      # Create an empty commit, which is set to be upstream
+      system "git", "commit", "--allow-empty", "--message", "Empty commit"
+      mkdir ".git/refs/remotes/origin"
+      system "git rev-parse HEAD >.git/refs/remotes/origin/trunk"
```
Comment (carlocab) [inline Formula/spr.rb:37]:
```suggestion
      (testpath/"test-repo/.git/refs/remotes/origin/trunk").atomic_write Utils.git_head
```

File: Formula/spr.rb (hunk 1 above)
Comment (sven-of-cord) [reply Formula/spr.rb:6]: I copied this from some other formula and didn't think much about it. But I guess it's cheaper to download a tarball than clone a Git repo.

Changing this.

File: Formula/spr.rb (hunk 2 above)
Comment (sven-of-cord) [reply Formula/spr.rb:27]: :+1:

**Comment by sven-of-cord** [comment]
Thanks for the feedback @carlocab. I made all the requested changes.

**Review by carlocab — DISMISSED** [review]
LGTM. We can merge this when CI is done. Thanks for the contribution!

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**